include/poffin.h

```diff
@@ -1,11 +1,31 @@
 #ifndef POKEPLATINUM_POFFIN_H
 #define POKEPLATINUM_POFFIN_H
 
-#include "struct_decls/struct_0202AB28_decl.h"
-#include "struct_defs/struct_0202A93C.h"
-
 #include "savedata.h"
 
+#define POFFIN_NONE 0xFFFF
+#define MAX_POFFINS 100
+
+typedef struct {
+    union {
+        struct {
+            u8 flavor;
+            u8 spiciness;
+            u8 dryness;
+            u8 sweetness;
+            u8 bitterness;
+            u8 sourness;
+            u8 val1_06;
+            u8 dummy;
+        };
+        u8 attributes[8];
+    };
+} Poffin;
```

Comment:
Would something like this work?

```suggestion
typedef struct {
    union {
        struct PoffinInner {
            u8 flavor;
            u8 spiciness;
            u8 dryness;
            u8 sweetness;
            u8 bitterness;
            u8 sourness;
            u8 val1_06;
            u8 dummy;
        };
        u8 attributes[sizeof(struct PoffinInner)];
    };
} Poffin;
```

This way, the size of `attributes` is always tied to the size of the inner struct used in the union.

Reply:
We're about to find out